src/plugins/visualizations/public/react_embeddable/visualize_embeddable.tsx

```diff
@@ -0,0 +1,532 @@
+
+                  let telemetryVisTypeName = visTypeName;
+                  if (visTypeName === 'metrics') {
+                    telemetryVisTypeName = 'legacy_metric';
+                  }
+                  if (visTypeName === 'pie' && visInstance.params.isDonut) {
+                    telemetryVisTypeName = 'donut';
+                  }
+                  if (
+                    visTypeName === 'area' &&
+                    visInstance.params.seriesParams.some(
+                      (seriesParams: { mode: string }) => seriesParams.mode === 'stacked'
+                    )
+                  ) {
+                    telemetryVisTypeName = 'area_stacked';
+                  }
+
+                  getUsageCollection().reportUiCounter(
+                    executionContext?.type ?? '',
+                    'count',
+                    `render_agg_based_${telemetryVisTypeName}`
+                  );
+
+                  if (hasRendered$.getValue() === true) return;
+                  hasRendered$.next(true);
+                  hasRendered$.complete();
+                },
+                onEvent: async (event) => {
+                  // Visualize doesn't respond to sizing events, so ignore.
+                  if (isChartSizeEvent(event)) {
+                    return;
+                  }
+                  const currentVis = vis$.getValue();
+                  if (!disableTriggers) {
+                    const triggerId = get(
+                      VIS_EVENT_TO_TRIGGER,
+                      event.name,
+                      VIS_EVENT_TO_TRIGGER.filter
+                    );
+                    let context;
+
+                    if (triggerId === VIS_EVENT_TO_TRIGGER.applyFilter) {
+                      context = {
+                        embeddable: api,
+                        timeFieldName: currentVis.data.indexPattern?.timeFieldName!,
+                        ...event.data,
+                      };
+                    } else {
+                      context = {
+                        embeddable: api,
+                        data: {
+                          timeFieldName: currentVis.data.indexPattern?.timeFieldName!,
+                          ...event.data,
+                        },
+                      };
+                    }
+                    await getUiActions().getTrigger(triggerId).exec(context);
+                  }
+                },
+                onData: (_, inspectorAdapters) => {
+                  inspectorAdapters$.next(
+                    typeof inspectorAdapters === 'function'
+                      ? inspectorAdapters()
+                      : inspectorAdapters
+                  );
+                  dataLoading$.next(false);
+                },
+              });
+              if (params) expressionParams$.next(params);
+              expressionAbortController$.next(abortController);
+            };
+            return await updateExpressionParams();
```

Comment:
`updateExpressionParams` should be changed to `getExpressionParams` and return `{ params, abortController }`.

Then this subscription should get these values and call `next`. The reason being, is that a switchMap will cancel the previous observable and create a new observable when the observable its attached to fires. This means that we should not be calling `next` until outside of the switchMap so that cancelled instances to update state if there is a race condition.